nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1ClientConnection.java

```diff
@@ -171,16 +174,32 @@ public void writeNow(BufferData buffer) {
 
     void finishRequest() {
         if (keepAlive && connectionQueue != null && socket.isConnected()) {
-            if (connectionQueue.offer(this)) {
+            try {
+                if (connectionQueue.offer(this, QUEUE_TIMEOUT, QUEUE_TIMEOUT_TIME_UNIT)) {
+                    if (LOGGER.isLoggable(DEBUG)) {
```

Comment:
Using the supplier version hides these if statements and makes the code arguably more readable.

Reply:
Santiago mentioned`the Logger should have a version of the method that takes a lambda so you don't need to check the level with an if-statement.`, so will look into this.